server/services/v1/api.go

```diff
@@ -411,19 +411,39 @@ func (s *apiService) Events(r *api.EventsRequest, stream api.Tigris_EventsServer
 	if !config.DefaultConfig.Cdc.Enabled {
 		return api.Errorf(api.Code_METHOD_NOT_ALLOWED, "change streams is disabled for this collection")
 	}
+
 	publisher := s.cdcMgr.GetPublisher(r.GetDb())
 	streamer, err := publisher.NewStreamer(s.kvStore)
 	if err != nil {
 		return err
 	}
 	defer streamer.Close()
 
+	reqDatabaseId, reqCollectionId := uint32(0), uint32(0)
 	for tx := range streamer.Txs {
 		for _, op := range tx.Ops {
-			_, _, collection, ok := s.tenantMgr.DecodeTableName(op.Table)
+			if reqDatabaseId == 0 || reqCollectionId == 0 {
+				if reqDatabaseId, reqCollectionId = s.tenantMgr.GetDatabaseAndCollectionId(r.GetDb(), r.GetCollection()); reqDatabaseId == 0 || reqCollectionId == 0 {
+					// neither is ready yet
+					continue
+				}
+			}
+
+			nsId, dbId, cId, ok := s.tenantMgr.GetEncoder().DecodeTableName(op.Table)
+			if !ok {
+				log.Err(err).Str("table", string(op.Table)).Msg("unexpected key in event streams")
+				return api.Errorf(api.Code_INTERNAL, "unexpected key in event streams")
+			}
+
+			if dbId != reqDatabaseId || cId != reqCollectionId {
+				// this probably means database/collection is dropped and this entry is some old entry in the log, ignore it
+				continue
+			}
+
+			_, _, collection, ok := s.tenantMgr.GetTableNameFromIds(nsId, dbId, cId)
```

Comment:
We don't need collection name check here and below anymore.
ids of the event is equal to cached ids, so it must an event for the collection we are listening to